### PR TITLE
chore(monorepo): replace lerna with turbo

### PR DESCRIPTION
**Pull request by schultzp2020**
## Description

Please explain the changes you made here.

- Adds turborepo as the monorepo tool since lerna was not used in any scripts
  - Faster script runtime because of caching and parallelism
  - We also use turborepo on the showcase
- ~~Removes the `package` folder~~
- ~~Renames the `plugins` folder as `packages~~
  - ~~Add missing dependencies as dev dependencies in `package.json` to pass tests~~
  - ~~Inspiration from the [GitLab plugins](https://github.com/immobiliare/backstage-plugin-gitlab)~~

## Which issue(s) does this PR fix

- Blocked by https://github.com/janus-idp/backstage-plugins/pull/367

- Fixes build issues in the downstream

### CODE REVIEW

**Comment by nickboldt** [comment]
Well, it takes over an hour, but it passed!

https://orch.psi.redhat.com/pnc-web/#/projects/1631/build-configs/11003/builds/AYQSUI34TRQAA

File: turbo.json
Comment (RobotSail) [inline turbo.json:14]: Needs to depend on `build`
```suggestion
    "test": {
      "dependsOn": ["^build"]
    }  
```

File: turbo.json
Comment (RobotSail) [inline turbo.json:6]: Needs to depend on build
```suggestion
      "persistent": true,
      "dependsOn": ["^build"]
```

File: app-config.yaml
Comment (RobotSail) [inline app-config.yaml:1]: Why are we deleting this file?

File: package.json
Comment (RobotSail) [inline package.json:11]: Is this part of `build`? 

If not, is it possible to break this out so that all other plugins also have a `tsc` step, and that way we can just replace this with `turbo tsc`?

File: .dockerignore
Comment (RobotSail) [inline .dockerignore:1]: Why is this being deleted?

**Review by RobotSail — CHANGES_REQUESTED** [review]
Looks okay to me for the most part apart from some minor changes. 
Would like to get @tumido to look at this as well before merging

File: .dockerignore
Comment (schultzp2020) [reply .dockerignore:1]: There are no dockerfiles

File: app-config.yaml
Comment (schultzp2020) [reply app-config.yaml:1]: `app-config.yaml` is only needed for running an instance of backstage, not for the plugins.

File: package.json
Comment (schultzp2020) [reply package.json:11]: I can look into this, the reason I never looked into this is because TS tooling can be a pain in the butt.

File: .dockerignore
Comment (RobotSail) [reply .dockerignore:1]: Gotcha

File: package.json
Comment (schultzp2020) [reply package.json:11]: I don't think its worth changing because it actually takes longer to build on average and it adds extra boilerplate to each plugin. 

Here are my averages after some testing: 
| Before | After  |
| ------- | ------ |
| 11.5s   | 13.1s  |

Comment (RobotSail) [reply package.json:11]: Okay, that's fine, we can just leave it as is then.

**Review by RobotSail — COMMENTED** [review]
Turbo file looks good to me. 
Would like to hear @tumido's thoughts on these changes as well.

**Review by tumido — CHANGES_REQUESTED** [review]
So far I have just a brief look at the PR. Overall I love it, I would love to make this change happen and embrace life without `packages/[backend,app]`. It would simplify our lifes a lot.

However as of now I have to block this change from happening from a simple reason. Without `packages/backend` we have no way to develop catalog based plugins. We have no way to verify ingestion into catalog. This essentially blocks any meaningfull development and validation on `keycloak`, `3scale` and `ocm` plugins. 

I'll try to come up with a way how we can stand up a standalone catalog dev environments for those plugins. Without that we can't remove catalog (`packages/backend`) from the repo.

I've filed this newly discovered limitaion as #330 issue.

**Comment by nickboldt** [comment]
>  I have to block this change from happening

This means that for now I have to build RHPIB from Paul's fork (this PR)... not ideal for a GA, but OK for a dev preview. We will need a full solution before we can GA, so please treat this as a SEV1 / P1 / blocker (whatever terminology you want). :D

**Comment by tumido** [comment]
I'll try to resolve this next week. I know this is important to get in.

**Comment by rtaniwa** [comment]
hey @tumido . i think we may need to unblock this and merge, and then possibly address the other concerns later. i am worried that we are holding up our ability to GA RHPIB and we currently have no one available to pick up issue #330 unless you have other options.

**Comment by RobotSail** [comment]
@rtaniwa I am currently working on issue #330

**Review by tumido — CHANGES_REQUESTED** [review]
Please rebase properly. This PR diff includes code changes in one of the plugins, documentation changes in others, it's a mess. Also, please leave the `app-config*.yaml` files intact

**Review by tumido — APPROVED** [review]
/lgtm
/approve

**Review by tumido — APPROVED** [review]
/lgtm
/approve

**Pull request merged**
